Updated docs to add section on fail-over to production and staging ennvironment / resolve refs in dowstream projects (#6317)
fishtown-analytics/docs.getdbt.com: website/docs/docs/collaborate/govern/project-dependencies.md

- Use a supported version of dbt (v1.6, v1.7, or go versionless with "[Versionless](/docs/dbt-versions/upgrade-dbt-version-in-cloud#versionless)") for both the upstream ("producer") project and the downstream ("consumer") project.
- Define models in an upstream ("producer") project that are configured with [`access: public`](/reference/resource-configs/access). You need at least one successful job run after defining their `access`.
- Define a deployment environment in the upstream ("producer") project [that is set to be your Production environment](/docs/deploy/deploy-environments#set-as-production-environment), and ensure it has at least one successful job run in that environment.

> - Execute a job in the staging environment of the upstream project in order for the downstream cross-project ref to resolve.

Reviewer: recommend adding an 'if the upstream has a staging env'. otherwise it reads that the user _has_ to run a job in staging but what if they don't have a staging env?

```suggestion
- If the upstream project has a Staging environment, run a job in that Staging environment to ensure the downstream cross-project ref resolves.
```
